app/controllers/authenticated_controller.rb

```diff
@@ -47,5 +42,17 @@ def user_id
     current_user.id
   end
 
+  def check_roles!
+    allowed = allowed_roles.any? { |role| organisation&.has_role?(role) }
+
+    if allowed_roles.present? && !allowed
```

Comment:
What do you think about moving this logic to an `allowed?` method?

Reply:
Good idea! Have migrated to that